src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -38,7 +39,8 @@ public PipelineCacheServer(AgentTaskPluginExecutionContext context)
             ContentFormat contentFormat)
         {
             VssConnection connection = context.VssConnection;
-            var (dedupManifestClient, clientTelemetry) = await DedupManifestArtifactClientFactory.Instance.CreateDedupManifestClientAsync(context, connection, cancellationToken);
+            var (dedupManifestClient, clientTelemetry) = await DedupManifestArtifactClientFactory.Instance
+                .CreateDedupManifestClientAsync(context.IsSystemDebugTrue(), (str) => context.Output(str), connection, cancellationToken);
```

Comment:
Is this the bit that would print hashes?

Reply:
Also the context.IsSystemDebugTrue() seems to be a constant check, is it an option to do it in CreateDedupManifestClientAsync and just send the context?

Reply:
The "context" is different when called from a plugin or a command. `IsSystemDebugTrue` only exists on the plugin context.